crates/polars-plan/src/logical_plan/file_scan.rs

```diff
@@ -16,7 +16,13 @@ pub enum FileScan {
         metadata: Option<Arc<FileMetaData>>,
     },
     #[cfg(feature = "ipc")]
-    Ipc { options: IpcScanOptions },
+    Ipc {
+        options: IpcScanOptions,
+        #[cfg(feature = "cloud")]
+        cloud_options: Option<CloudOptions>,
+        #[cfg_attr(feature = "serde", serde(skip))]
+        metadata: Option<arrow::io::ipc::read::FileMetadata>,
```

Comment:
Can't make this `FileMetadata` (without the `Option`) because the struct implements `Deserialize`. This means that I can not statically guarantee that the metadata was already loaded.

Are we able to deserialize logical execution plans and execute them? Should we include the metadata in the serialized form?

Reply:
The actual reader can accept an `Option<Metdata>` and if `None` it must fetch it again.

Reply:
Regardless, this is not something that we'll address in this PR probably so I am resolving this conversation.